modules/sitemaps/sitemap-finder.php

```diff
@@ -0,0 +1,138 @@
+<?php
+/**
+ * The functions in this class provide an API for handling
+ * sitemap related URIs.
+ *
+ * @package Jetpack
+ * @since 4.5.0
+ * @author Automattic
+ */
+
+/**
+ * The Jetpack_Sitemap_Finder object deals with constructing
+ * sitemap URIs.
+ *
+ * @since 4.5.0
+ */
+class Jetpack_Sitemap_Finder {
+
+	/**
+	 * Jetpack_Sitemap_Finder constructor.
+	 *
+	 * @access public
+	 * @since 4.5.0
+	 */
+	public function __construct() {
+		return;
+	}
+
+	/**
+	 * Construct the complete URL of a sitemap file. Depends on
+	 * permalink settings.
+	 *
+	 * @access public
+	 * @since 4.5.0
+	 *
+	 * @param string $filename The filename of the sitemap.
+	 */
+	public function construct_sitemap_url( $filename ) {
+		global $wp_rewrite;
+
+		$location = get_option( 'jetpack_sitemap_location', '' );
```

Comment:
Will this benefit from using `Jetpack_Options::get_option_and_ensure_autoload( 'jetpack_sitemap_location', '' )`?
How about others `get_option` calls in this PR?